### PR TITLE
Chapter11

### PR DESCRIPTION
**Pull request by NurAnisAqilah**
ready

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
ready

File: app/assets/stylesheets/account_activations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/account_activations.scss]: remove file tha t never used

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,67 +1,89 @@
...
+  
+  def show
+    @user = User.find_by id: params[:id]
+  end 
+    flash[:danger] = "User not found!"
+    redirect_to root_path
 
   def index
-    @users = User.page(params[:page]).per Settings.paging
+      @users = User.page(params[:page]).per Settings.per_page
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: indenting only 2 spaces

File: app/helpers/account_activations_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/account_activations_helper.rb]: remove file that never used

File: app/assets/stylesheets/account_activations.scss (hunk 1 above)
Comment (saobv-2223) [reply app/assets/stylesheets/account_activations.scss]: remove file that never used

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    @users = User.page(params[:page]).per Settings.per_page
```

**Review by saobv-2223 — COMMENTED** [review]
fix all the comments and ready again

**Comment by NurAnisAqilah** [comment]
ready

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
   before_action :correct_user, only: %i(edit update)
-  before_action :admin_user, only: :destroy
-  before_action :load_user, except: %i(index new create)
+  before_action :admin_user, only: :destroy 
+  
+  def show
+    @user = User.find_by id: params[:id]
+  end 
+    flash[:danger] = "User not found!"
+    redirect_to root_path
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def show
    @user = User.find_by id: params[:id]
    return if @user 
    
    flash[:danger] = "User not found!"
    redirect_to root_path
  end 
```

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -12,56 +18,72 @@ def new
     @user = User.new
   end
 
+  def show; end
+
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb:24]:
```suggestion
```

**Comment by NurAnisAqilah** [comment]
ready

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -12,56 +20,73 @@ def new
     @user = User.new
   end
 
+  def show;
+  end
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def show; end
```

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -12,56 +20,73 @@ def new
...
     else
+      flash.now[:danger] = t "user.update_fail"
       render :edit
-      flash[:danger] = "Profile update unsuccessful"
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: move edit action above update action

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -12,56 +20,73 @@ def new
...
   end
 
-  def correct_user
-    @user = User.find_by id: params[:id]
-    return if current_user?@user
-    
-    flash[:danger] = "Unauthorized entry"
-    redirect_to root_url
+  def load_user
+    @user  = User.find_by id: params[:id]
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    @user = User.find_by id: params[:id]
```

**Comment by NurAnisAqilah** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**